cmd/kubelet/app/server.go

```diff
@@ -423,8 +424,12 @@ func Run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 	return nil
 }
 
+// checkPermissions confirms that, on linux, the UID of the kubelet process is 0.  For windows, it will return an error automatically.  For other systems, the behaviour is unspecified.
```

Comment:
i agree that this check should not be in the checkPermissions() function, but instead be on the side of the caller.
if you want the log entry still, you could do:

```
if runtime.GOOS == "windows" {
	klog.Info("skipping permissions check on Windows")
} else {
	if err := checkPermissions(); err != nil {
		klog.Error(err)
	}
}
```